3party/harfbuzz/CMakeLists.txt

```diff
@@ -28,14 +37,20 @@ target_compile_definitions(${PROJECT_NAME}
     #$<$<BOOL:${APPLE}>:HAVE_CORETEXT>
     HAVE_ATEXIT
     HAVE_GETPAGESIZE
-    HAVE_MMAP
     HAVE_MPROTECT
-    HAVE_PTHREAD
     HAVE_SYSCONF
-    HAVE_SYS_MMAN_H
-    HAVE_UNISTD_H
```

Comment:
All these defines are used in the HB code. Consider using generator expressions like `$<NOT:$<CXX_COMPILER_ID:MSVC>>`